### PR TITLE
Update Envoy

### PR DESCRIPTION
**Pull request by i0rek**
The new envoy versions are fixing a couple of CVEs: https://www.envoyproxy.io/docs/envoy/latest/version_history/v1.14.4.

## Checklist

- [x] Run the full Envoy integration test using the new versions:
```
make test-envoy-integ ENVOY_VERSIONS="1.14.4"
make test-envoy-integ ENVOY_VERSIONS="1.13.4"
make test-envoy-integ ENVOY_VERSIONS="1.12.6"
```

- [x] Update `ENVOY_VERSION` in `test/integration/connect/envoy/run-tests.sh` to the latest version
- [x] Update the CI jobs in .circleci/config.yml
  - [x] There are multiple jobs named envoy-integration-test-* where the oldest has a full definition while the others are aliases that just change the version number.
  - [x] Update the list of envoy-integration-test-* jobs in the test-integrations workflow to match the new job names changes above.
- [x] Update `-envoy-version` to the latest version in `website/pages/docs/commands/connect/envoy.mdx` and `defaultEnvoyVersion` in `command/connect/envoy/envoy.go`
- [x] Update `agent/xds/clusters_test.go: supportedEnvoyVersions`
- [x] Update the supported versions list in the docs in `website/pages/docs/connect/proxies/envoy.mdx`
- [x] Regenerate `testdata`:
  - [x] `go test ./command/connect/envoy -update`
  - [x] `go test ./agent/xds -update`
- [x] Create an issue consul-k8s to update the default Envoy image to be the latest supported version. ~~It's currently here: https://github.com/hashicorp/consul-k8s/blob/f75b6f559fae9f96667820796edd47c0c7a3824b/connect-inject/handler.go#L23 but probably worth checking the code base and updating this list if there are other references:~~ Consul 1.8.1 will ship with support for envoy 1.14.4: https://github.com/hashicorp/consul-k8s/issues/287
 - [x] Create an issue for consul-helm to update the default Envoy image. ~~Currently https://github.com/hashicorp/consul-helm/blob/master/values.yaml#L609 but again grep and sanity check~~: https://github.com/hashicorp/consul-helm/issues/525.
 - [x] Create an issue for Nomad to update the default Envoy docker image they use in their integration with a link to the PR and information about the version we will make the change in and the expected timeline for release: https://github.com/hashicorp/nomad/issues/7665#issuecomment-653516276.

### CODE REVIEW

**Comment by i0rek** [comment]
Fails with `envoyproxy/envoy:v1.12.5` atm.

**Comment by i0rek** [comment]
https://github.com/envoyproxy/envoy/issues/11838

**Comment by rboyer** [comment]
On 7/8 `v1.14.4`, `v1.13.4`, and `v1.12.6` are now available.

**Comment by rboyer** [comment]
This should probably wait until the https://github.com/hashicorp/consul/pull/8222 (and backports) land.

File: agent/xds/clusters_test.go [hunk 1]
```diff
@@ -742,8 +742,8 @@ func setupTLSRootsAndLeaf(t *testing.T, snap *proxycfg.ConfigSnapshot) {
 //
 // see: https://www.consul.io/docs/connect/proxies/envoy#supported-versions
 var supportedEnvoyVersions = []string{
```
Comment (rboyer) [inline agent/xds/clusters_test.go:744]: This is new with the new version sniffing logic. I rebased this entire PR.

**Comment by rboyer** [comment]
I've rebased the PR on my version sniffing PR and also bumped all of the versions to the extra double plus latest patch releases.

**Pull request merged**